### PR TITLE
Fix/aimperformance

### PR DESCRIPTION
**Pull request by emelalkim**

### CODE REVIEW

File: plugins/EpadDB.js [hunk 1]
```diff
@@ -4480,33 +4481,42 @@ async function epaddb(fastify, options, done) {
 
   fastify.decorate(
```
Comment (emelalkim) [inline plugins/EpadDB.js:4482]: remove unused method

File: plugins/EpadDB.js [hunk 2]
```diff
@@ -4813,28 +4837,43 @@ async function epaddb(fastify, options, done) {
           )
         );
       else {
-        const projectAimCount = await models.project_aim.count({
-          where: { project_id: project.id, aim_uid: request.params.aimuid },
+        let qry = `SELECT count(*) as count
```
Comment (emelalkim) [inline plugins/EpadDB.js:4840]: replace with new getaimsinternal?

File: plugins/EpadDB.js [hunk 3]
```diff
@@ -5698,8 +5745,10 @@ async function epaddb(fastify, options, done) {
               reply
                 .code(200)
                 .send(`Aim deleted from system as it didn't exist in any other project`);
-            } else
+            } else {
```
Comment (emelalkim) [inline plugins/EpadDB.js:5748]: delete project and deleteRelationAndOrphanedCouchDocInternal should make sure to update the project_ids in couchdb when deleted

**Review by OzgeYurtsever — APPROVED** [review]
Resolve issues.

File: plugins/CouchDB.js [hunk 4]
```diff
@@ -525,36 +689,40 @@ async function couchdb(fastify, options) {
   // filter aims with aimId filter array
   fastify.decorate(
     'filterAims',
-    (aimsRows, filter, format, params, epadAuth) =>
+    (aimsRows, filter) =>
```
[the remark this answers is not in the document]
Comment (emelalkim) [reply plugins/CouchDB.js:692]: done

File: plugins/CouchDB.js [hunk 5]
```diff
@@ -525,36 +689,40 @@ async function couchdb(fastify, options) {
...
           }
+          resolve(filteredRows);
+        } catch (err) {
+          reject(new InternalError('Filtering aims', err));
+        }
+      })
+  );
+
+  // TODO attempt for json filtering for getAims
+  fastify.decorate(
```
[the remark this answers is not in the document]
Comment (emelalkim) [reply plugins/CouchDB.js:709]: done

File: plugins/CouchDB.js [hunk 6]
```diff
@@ -671,6 +881,77 @@ async function couchdb(fastify, options) {
...
+          const aimsNotSaved = [];
+          for (let i = 0; i < aimsWithProjects.length; i += 1) {
+            try {
+              let noChange = false;
+              // eslint-disable-next-line no-await-in-loop
+              const couchDoc = await db.get(aimsWithProjects[i].aim);
+              if (couchDoc.projects) {
+                if (!couchDoc.projects.includes(aimsWithProjects[i].project))
+                  couchDoc.projects.push(aimsWithProjects[i].project);
+                noChange = true;
```
[the remark this answers is not in the document]
Comment (emelalkim) [reply plugins/CouchDB.js:901]: done

File: plugins/CouchDB.js [hunk 7]
```diff
@@ -671,6 +881,77 @@ async function couchdb(fastify, options) {
...
+                  }`
+                );
+              } else {
+                fastify.log.info(`No update for aimuid ${aimsWithProjects[i].aim}`);
+              }
+
+              editCount += 1;
+            } catch (err) {
+              fastify.log.error(`Error in saving aim ${aimsWithProjects[i].aim} to couchdb`, err);
+              // wait for couchdb to get healthy
```
[the remark this answers is not in the document]
Comment (emelalkim) [reply plugins/CouchDB.js:932]: done

File: plugins/EpadDB.js (hunk 1 above)
Comment (emelalkim) [reply plugins/EpadDB.js:4482]: done

File: plugins/EpadDB.js (hunk 2 above)
Comment (emelalkim) [reply plugins/EpadDB.js:4840]: done. getaimsinternal handles the user check now. using just that is enough

**Pull request merged**